### PR TITLE
backend: move balance to coin pkg, improve coin interfaces, rename init

### PR DESCRIPTION
**Pull request by KasparEtter**

### CODE REVIEW

File: backend/coins/coin/account.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+package coin
+
+import (
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/coin/fee"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/signing"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable"
+)
+
+// Account contains transfers of the given coin.
+type Account interface {
```
Comment (benma) [inline backend/coins/coin/account.go]: Easier to step-by-step move the btc.Interface into a state where it is coin agnostic, and then just move it? What I did with my previous commits, so it is easy to follow.

File: backend/backend.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ type AccountEvent struct {
 	Data string `json:"data"`
 }
 
-// Backend ties everything together and is the main starting point to use the BitBox wallet library.
+// Backend manages the devices, coins and accounts and exposes an API including events.
```
Comment (KasparEtter) [inline backend/backend.go]: TODO: Include with application.go

File: backend/coins/btc/coin.go [hunk 3]
```diff
@@ -124,14 +124,24 @@ func (coin *Coin) Unit() string {
...
-		new(big.Rat).SetFrac(amount.BigInt(), big.NewInt(unitSatoshi)).FloatString(8),
+		new(big.Rat).SetFrac(amount.BigInt(), coin.Denomination()).FloatString(8),
 		"0.",
 	)
 }
 
+// AccountBased implements coin.Coin.
+func (coin *Coin) AccountBased() bool {
+	return false
+}
```
Comment (KasparEtter) [inline backend/coins/btc/coin.go]: TODO: Include with a check in the frontend on the receive address screen.

Comment (KasparEtter) [reply backend/coins/btc/coin.go]: Or rather to determine whether coin selection is available in send screen.

**Pull request closed without merging**